server/src/main/java/org/opensearch/action/admin/indices/tiering/TransportHotToWarmTieringAction.java

```diff
@@ -92,19 +100,41 @@ protected void clusterManagerOperation(
     ) throws Exception {
         Index[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
         if (concreteIndices == null || concreteIndices.length == 0) {
+            logger.info("[HotToWarmTiering] No concrete indices resolved for the indices {} in the request", List.of(request.indices()));
             listener.onResponse(new HotToWarmTieringResponse(true));
             return;
         }
+        final Set<Index> hotIndices = new HashSet<>();
+        for (Index index : concreteIndices) {
+            if (isHotIndex(clusterService.state().metadata().getIndexSafe(index))) {
+                hotIndices.add(index);
+            }
+        }
+        if (hotIndices.isEmpty()) {
+            logger.info("[HotToWarmTiering] No hot indices found out of the resolved concrete indices {}", List.of(concreteIndices));
+            listener.onResponse(new HotToWarmTieringResponse(true));
+            return;
+        }
+
         final TieringValidationResult tieringValidationResult = validateHotToWarm(
             state,
-            Set.of(concreteIndices),
+            hotIndices,
             clusterInfoService.getClusterInfo(),
             diskThresholdSettings
         );
 
         if (tieringValidationResult.getAcceptedIndices().isEmpty()) {
-            listener.onResponse(tieringValidationResult.constructResponse());
+            listener.onResponse(constructToHotToWarmTieringResponse(tieringValidationResult.getRejectedIndices()));
             return;
         }
+
+        final TieringUpdateClusterStateRequest updateClusterStateRequest = new TieringUpdateClusterStateRequest(
```

Comment:
Here are the reasons of having `TieringUpdateClusterStateRequest` extends `IndicesClusterStateUpdateRequest` -
- All the other transport classes that have cluster state request on multiple indices follow this pattern(e.g. `OpenIndexClusterStateUpdateRequest`, `CloseIndexClusterStateUpdateRequest`)
- Since `TieringRequestContext` is cached in the tiering service, it doesnt need to actually need to store the actual request. 
- Request can have any other parameters passed in that could be part of `TieringUpdateClusterStateRequest` but not actually stored in `TieringRequestContext`